dmutils/logging.py

```diff
@@ -138,15 +137,18 @@ def process_log_record(self, log_record):
         log_record['logType'] = "application"
 
         missing_keys = []
-        for attempt in range(self._format_message_missing_key_attempts):
+        for attempt in range(5):
             try:
                 log_record['message'] = log_record['message'].format(**log_record)
+
+                if missing_keys:
+                    logger.warning("Missing keys when formatting log message: {}".format(missing_keys))
```

Comment:
The reason I yanked the equivalent of this line down into an `else` clause is because I wanted to be absolutely certain that any `KeyError` we caught came from the original formatting attempt.

Reply:
It can't be in an else after you break out of the try, as that else will never happen.

Reply:
ahhhhhrrrrr... humf